relay-general/src/store/transactions/processor.rs

```diff
@@ -341,28 +341,32 @@ impl Processor for TransactionsProcessor<'_> {
                 .set_value(Some("<unlabeled transaction>".to_owned()))
         }
 
-        // Normalize transaction names for URLs and Sanitized transaction sources.
-        // This in addition to renaming rules can catch some high cardinality parts.
-        if matches!(
-            event.get_transaction_source(),
-            &TransactionSource::Url | &TransactionSource::Sanitized
-        ) && self.name_config.scrub_identifiers
-        {
-            let changed = scrub_identifiers(&mut event.transaction)?;
-            if changed && self.name_config.mark_scrubbed_as_sanitized {
-                let source = &mut event
-                    .transaction_info
-                    .get_or_insert_with(Default::default)
-                    .source;
-                source.set_value(Some(TransactionSource::Sanitized));
+        if self.name_config.scrub_identifiers {
+            // Normalize transaction names for URLs and Sanitized transaction sources.
+            // This in addition to renaming rules can catch some high cardinality parts.
+            if matches!(
+                event.get_transaction_source(),
+                &TransactionSource::Url | &TransactionSource::Sanitized
+            ) {
+                scrub_identifiers(&mut event.transaction)?;
+                if self.name_config.mark_scrubbed_as_sanitized {
+                    // TODO(iker): we also mark transaction as sanitized after
+                    // applying renaming rules. Once we remove this feature
+                    // flag, relay should only mark transactions as sanitized
+                    // once.
+                    event
+                        .transaction_info
+                        .get_or_insert_with(Default::default)
+                        .source
+                        .set_value(Some(TransactionSource::Sanitized));
+                }
```

Comment:
It might not matter in practice, but I would put this block _after_ `apply_transaction_rename_rule`. Basically

1. Apply sanitation step 1
2. Apply sanitation step 2
3. Mark as sanitized

Right now the order is

1. Apply sanitation step 1
1. Mark as sanitized
1. Apply sanitation step 2

which feels unnatural to me.

Reply:
Docs of `mark_scrubbed_as_sanitized` are as follows:

> True if transaction names scrubbed by regex patterns should be marked as `TransactionSource::Sanitized`.
>
> Transaction names modified by clusterer rules are always marked as such.

So the code follows these steps:

1. Apply sanitation step 1.
2. Mark as sanitized, resulting from step 1.
3. Apply sanitation step 2 (which also marks as sanitized, so no `4` needed).

I think 1 and 2 belong together. I agree that this could be improved, and that's why I added the `TODO`, but I wanted to minimize refactoring as much as possible until we have more information on how this looks and what we do with the feature flags..

Reply:
> True if transaction names scrubbed by regex patterns should be marked as TransactionSource::Sanitized.

That doc was written before you removed the `if changed` condition though (by which it actually belonged to pattern-based matching). By removing that condition, the real consequence of `mark_scrubbed_as_sanitized` is that _every_ `url` transaction will be marked as sanitized as long as the `normalize-transaction-names` feature flag is on.